### ISSUE TITLE
⌨Add Alert to notify Screen reader users about Indigenous language compatibility

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/design-gc-digital-talent#10]
## ⌨ Accessibility Issue

There are no ISO language codes for some of the indigenous languages we are using. This means that the languages cannot be programmatically determined and we are failing the WCAG criteria below.

Some screen readers do not have settings to recognize some characters in our chosen Indigenous languages. The Screen reader is silent. To ensure acceptable pronunciation, we would have to test the Anishinaabemowin Western Ojibway with @MCommonda.  

### WCAG 2 Criteria

[3.1.1 Language of Page (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/language-of-page): The default [human language](https://www.w3.org/WAI/WCAG21/Understanding/language-of-page#dfn-human-language) of each Web page can be [programmatically determined](https://www.w3.org/WAI/WCAG21/Understanding/language-of-page#dfn-programmatically-determined).

[3.1.2 Language of Parts (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/language-of-parts): The human of each passage or phrase in the content can be programmatically determined except for proper names, technical terms, words of indeterminate language, and words or phrases that have become part of the vernacular of the immediately surrounding text.

[1.3.1 Info and relationships (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships): Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text.

## 🦋 Expected Behaviour

Indigenous language screen reader users should be notified somehow that their tools might not read the characters on their language home pages correctly.   

## 🕵️ Details

- **Operating System**: All operating systems
- **Browser**: All browsers
- **Assistive Technology**: Al screen readers

## 📋 Steps to Reproduce

1. Turn on any screen reader
2. Navigate to each Language homepage. 
3. Listen to what you hear.

## 🙋‍♀️ Proposed Solution

 - In the short term, we could take a similar approach and provide a notice to screen reader users that is similar to the[ National Research Council](https://nrc.canada.ca/en/research-development/research-collaboration/programs/fact-sheet-indigenous-languages-technology-project)
 - In the long term, we could explore a partnership with the [Indigenous Language Technology Project](https://www.bing.com/ck/a?!&&p=b75329833b463b2fJmltdHM9MTY5NTYwMDAwMCZpZ3VpZD0xODVkNWY3MC0wZjkwLTZlM2QtMGJhNy00YzBjMGU1MTZmZDAmaW5zaWQ9NTIzMg&ptn=3&hsh=3&fclid=185d5f70-0f90-6e3d-0ba7-4c0c0e516fd0&psq=indigenous+languages+technoilogy+project&u=a1aHR0cHM6Ly9ucmMtcHVibGljYXRpb25zLmNhbmFkYS5jYS9lbmcvdmlldy9vYmplY3QvP2lkPWQ0ZjEwMTQ0LWM3MTEtNDNjNS1iODBiLTVhY2U3ZGY1ZTY4Yg&ntb=1) (ILT) at the National Research Council to see if their technology experiments might be helpful to us.

## 📸 Screenshot

![Image](https://github.com/GCTC-NTGC/design-gc-digital-talent/assets/101275838/b0603905-5463-4f8f-9e58-cc14cbb796e9)

## ✅ Acceptance Criteria

- [ ] The notice appears at the top of the main content.

**Comment by gobyrne** [comment]
@RM-1978 I think this is a good recommendation, but I don't feel like it would be required under WCAG. At least not for the criteria identified in this issue or to my knowledge of others.

So I would suggest that @MCommonda bring this to the IAP team as a recommendation. But I'll give them final call, and a "no leave it off" is acceptable.

**Comment by MCommonda** [comment]
Not sure how old this issue is but I didn't receive a notification as my name was spelt incorrectly in the opening context. I would like to have a better idea how the screen reader is going to pronounce the text in Indigenous languages before bringing it to the IAP team.

**Comment by RM-1978** [comment]
Sounds good. Are you ok to test it with a screen reader, or would you like to have @esizer or I review the pages with you on ou screen reader?

**Comment by RM-1978** [comment]
> @RM-1978 I think this is a good recommendation, but I don't feel like it would be required under WCAG. At least not for the criteria identified in this issue or to my knowledge of others.
> 
> So I would suggest that @MCommonda bring this to the IAP team as a recommendation. But I'll give them final call, and a "no leave it off" is acceptable.

Thanks @Gray for your comment on WCAG 2. Please note that iti is nin combination with WCAG 3.1.1 and 3.1.2 that 1.3.1 become relevant. I will share your comments with my Design Accessibility instructor. the Senior Inclusive Designer at Microsoft.

**Comment by MCommonda** [comment]
I'd be happy to set up a testing session, however I am not a fluent language speaker, so I would not be in an appropriate position to identify whether or not the pronunciations are correct or not. happy to discuss more during testing session. May need to try and get a language speaker involved though.

**Comment by MCommonda** [comment]
@RM-1978 as a side note - I just did a quick review of the IAP homepage with the Narrator tool on the Anishinaabemowin page and although I'm not a fluent speaker, it is safe to say that there are quite a few mispronunciations.

**Comment by RM-1978** [comment]
@gobyrne - Thanks for chatting about the potential solution for this. The next steps would be to design the alert content for a screen reader to determine and then implement it using a hidden / screen reader-only message. If we designed the message for this sprint, then we could estimate the effort with the Devs for a future sprint.

**Comment by gobyrne** [comment]
We determined with the IAP team that the current approach is sufficient. Closing.

**Comment by RM-1978** [comment]
@gobyrne Could you please remind me of what the current approach is?

**Comment by gobyrne** [comment]
@RM-1978 The ISO codes were adding in some time ago (Matt was able to find the missing ones).

So the current approach is to just let the screen readers do their job and not add any further interventions for languages that they are not able to read. Our code meets the WCAG requirement for the languages being programmatically identifiable.

I don't expect these pages to come up in much further AT user tests, but if they do, we can decide how to handle them then.

**Comment by gobyrne** [comment]
#8214 is where we added the language codes
